File: kvt-projekat2024/src/app/models/Discipline.ts
import { Facility } from "./Facility";

export class Discipline {
  id: number;
  name: string;
  facility?: Facility;

  constructor(id: number, name: string, facility: Facility) {
    this.id = id;
    this.name = name;
    this.facility = facility;
  }
}

File: src/main/java/rs/ac/uns/ftn/svt/model/Discipline.java
package rs.ac.uns.ftn.svt.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
public class Discipline {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "facility_id", nullable = false)
    private Facility facility;

}

